### PR TITLE
Add GL_EXT_shader_subgroup_extended_types support

### PR DESCRIPTION
**Pull request by jeffbolznv**

### CODE REVIEW

File: glslang/MachineIndependent/Initialize.cpp [hunk 1]
```diff
@@ -1803,1002 +1751,130 @@ void TBuiltIns::initialize(int version, EProfile profile, const SpvVersion& spvV
...
-                "dvec4  subgroupPartitionedExclusiveMinNV(dvec4, uvec4 ballot);\n"
-
-                "double subgroupPartitionedExclusiveMaxNV(double, uvec4 ballot);\n"
-                "dvec2  subgroupPartitionedExclusiveMaxNV(dvec2, uvec4 ballot);\n"
-                "dvec3  subgroupPartitionedExclusiveMaxNV(dvec3, uvec4 ballot);\n"
-                "dvec4  subgroupPartitionedExclusiveMaxNV(dvec4, uvec4 ballot);\n"
-
-                "\n");
+        // Generate all flavors of subgroup ops.
+        static const char *subgroupOps[] =
```
Comment (johnkslang) [inline glslang/MachineIndependent/Initialize.cpp]: Nice.

File: glslang/MachineIndependent/Initialize.cpp [hunk 2]
```diff
@@ -1803,1002 +1751,130 @@ void TBuiltIns::initialize(int version, EProfile profile, const SpvVersion& spvV
...
+            "uint8_t", "u8vec2", "u8vec3", "u8vec4", 
+            "uint16_t", "u16vec2", "u16vec3", "u16vec4", 
+            "uint", "uvec2", "uvec3", "uvec4", 
+            "uint64_t", "u64vec2", "u64vec3", "u64vec4", 
+        };
+        static const char *boolTypes[] = { 
+            "bool", "bvec2", "bvec3", "bvec4", 
+        };
+
+        for (unsigned int i = 0; i < sizeof(subgroupOps)/sizeof(subgroupOps[0]); ++i) {
```
Comment (johnkslang) [inline glslang/MachineIndependent/Initialize.cpp]: This needs to be `size_t` to satisfy some environments.

File: glslang/MachineIndependent/Initialize.cpp [hunk 3]
```diff
@@ -1803,1002 +1751,130 @@ void TBuiltIns::initialize(int version, EProfile profile, const SpvVersion& spvV
...
+            bool logicalOp = strstr(op, "Or") || strstr(op, "And") ||
+                             (strstr(op, "Xor") && !strstr(op, "ShuffleXor"));
+            // Math operations don't support bool
+            bool mathOp = strstr(op, "Add") || strstr(op, "Mul") || strstr(op, "Min") || strstr(op, "Max");
+
+            const int bufSize = 256;
+            char buf[bufSize];
+
+            if (!logicalOp) {
+                for (unsigned int j = 0; j < sizeof(floatTypes)/sizeof(floatTypes[0]); ++j) {
```
Comment (johnkslang) [inline glslang/MachineIndependent/Initialize.cpp]: `size_t`

File: glslang/MachineIndependent/Initialize.cpp [hunk 4]
```diff
@@ -1803,1002 +1751,130 @@ void TBuiltIns::initialize(int version, EProfile profile, const SpvVersion& spvV
...
+                    snprintf(buf, bufSize, op, floatTypes[j], floatTypes[j]);
+                    commonBuiltins.append(buf);
+                }
+                if (profile != EEsProfile && version >= 400) {
+                    for (unsigned int j = 0; j < sizeof(doubleTypes)/sizeof(doubleTypes[0]); ++j) {
```
Comment (johnkslang) [inline glslang/MachineIndependent/Initialize.cpp]: `size_t`

**Comment by jeffbolznv** [comment]
Fixed the size_t stuff.

**Pull request merged**